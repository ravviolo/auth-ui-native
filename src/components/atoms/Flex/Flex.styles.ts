import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  column: {
    flexDirection: 'column',
  },
  row: {
    flexDirection: 'row',
  },
});
