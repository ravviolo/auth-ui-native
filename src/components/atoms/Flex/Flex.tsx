import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';

export interface Props {
  children: React.ReactNode;
  direction: 'column' | 'row';
  testID: string;
  style?: StyleProp<ViewStyle>;
}

export const Flex = ({ children, direction, style, testID }: Props) => {
  return (
    <View style={[styles[direction], style]} testID={testID}>
      {children}
    </View>
  );
};

const styles = StyleSheet.create({
  column: {
    flexDirection: 'column',
  },
  row: {
    flexDirection: 'row',
  },
});
