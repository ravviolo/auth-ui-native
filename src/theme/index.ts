interface Colors {
  text: string;
  textLight: string;
  textWhite: string;
  primary: string;
  primaryDark: string;
  background: string;
}

export const colors: Colors = {
  text: '#3D3D3D',
  textLight: '#9E9E9E',
  textWhite: '#EDEDED',
  primary: '#FF4684',
  primaryDark: '#F9256C',
  background: '#fff',
};
