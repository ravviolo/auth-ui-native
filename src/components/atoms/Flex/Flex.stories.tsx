import { select, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { View } from 'react-native';
import { StoryScreen } from 'storybook/StoryScreen';
import { colors } from 'theme';

import { Flex } from './Flex';

const FlexChildren = Array(10)
  .fill(1)
  .map((_, i) => (
    <View
      // eslint-disable-next-line react/no-array-index-key
      key={i}
      style={{
        width: 10,
        height: 10,
        margin: 10,
        backgroundColor: colors.primary,
      }}
    />
  ));

const flexOptions: ['column', 'row'] = ['column', 'row'];

storiesOf('Flex', module)
  .addDecorator((getStory) => (
    <StoryScreen style={{ alignItems: 'center' }}>{getStory()}</StoryScreen>
  ))
  .add('Column', () => (
    <Flex
      direction={select('direction', flexOptions, 'column')}
      testID={text('testID', 'flex-test-id')}
    >
      {FlexChildren}
    </Flex>
  ))
  .add('Row', () => (
    <Flex
      direction={select('direction', flexOptions, 'row')}
      testID={text('testID', 'flex-test-id')}
    >
      {FlexChildren}
    </Flex>
  ));
