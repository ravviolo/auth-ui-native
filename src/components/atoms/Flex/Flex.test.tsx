import { render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';
import { colors } from 'theme';

import { Flex } from './Flex';

import type { Props } from './Flex';

describe('Flex', () => {
  const props: Omit<Props, 'direction'> = {
    testID: 'flex-test-id',
    children: Array(10)
      .fill(1)
      .map((_, i) => (
        <li
          // eslint-disable-next-line react/no-array-index-key
          key={i}
        />
      )),
  };

  it('should render all children horizontally', () => {
    render(<Flex {...props} direction="row" />);

    const flexContainer = screen.getByTestId('flex-test-id');

    expect(flexContainer.children).toHaveLength(10);
    expect(flexContainer).toHaveStyle({ flexDirection: 'row' });
  });

  it('should render all children vertically', () => {
    render(<Flex {...props} direction="column" />);

    const flexContainer = screen.getByTestId('flex-test-id');

    expect(flexContainer.children).toHaveLength(10);
    expect(flexContainer).toHaveStyle({ flexDirection: 'column' });
  });

  it('should apply additional styles from props', () => {
    render(<Flex {...props} direction="column" style={{ backgroundColor: colors.primary }} />);

    const flexContainer = screen.getByTestId('flex-test-id');
    expect(flexContainer).toHaveStyle({ backgroundColor: colors.primary });
  });

  it('should match snapshot - render all children horizontally', () => {
    const tree = testRenderer.create(<Flex {...props} direction="row" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot - render all children vertically', () => {
    const tree = testRenderer.create(<Flex {...props} direction="column" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot - apply styles', () => {
    const tree = testRenderer
      .create(<Flex {...props} direction="row" style={{ backgroundColor: colors.primary }} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
