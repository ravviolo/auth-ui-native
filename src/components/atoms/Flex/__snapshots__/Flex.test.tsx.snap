// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex should match snapshot - apply styles 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "#FF4684",
      },
    ]
  }
  testID="flex-test-id"
>
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
</View>
`;

exports[`Flex should match snapshot - render all children horizontally 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
  testID="flex-test-id"
>
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
</View>
`;

exports[`Flex should match snapshot - render all children vertically 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "column",
      },
      undefined,
    ]
  }
  testID="flex-test-id"
>
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
</View>
`;
